samples/radio_test_bm/src/nrf_wifi_radio_test_shell.c

```diff
@@ -996,6 +1037,7 @@ static int nrf_wifi_radio_test_set_tx(size_t argc, const char *argv[]) {
 
   if (status != NRF_WIFI_STATUS_SUCCESS) {
     RT_SHELL_PRINTF_ERROR("Programming TX failed\n");
+    ctx->conf_params.tx = !val;
```

Comment:
This might cause an issue where a test is NOT running and the user issues a `TX stop` (erroneously). We seem to be missing a check in the beginning to catch this scenario. If now the `nrf_wifi_rt_fmac_prog_tx` too fails then the state will get set to `TX running`.

Reply:
it will still work, as the API won't fail if the test is not running, no? (we can add a check independent to this).

Reply:
I tried to do `wifi_radio_test tx_tone 0` without starting and the command timeout, like most other commands FW doesn't send an event if its a no-op, so, we need a fix in OSAL code for that to avoid sending CMD if it's a no-op, but this is unrelated to the issue that customer has reported.